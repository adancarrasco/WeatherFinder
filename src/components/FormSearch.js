import React, { Component } from "react";

import { fetchWeatherByCityName } from "../utils/weatherSvs";

import styles from "./FormSearch.module.css";

class FormSearch extends Component {
  constructor(props) {
    super(props);
    this.state = {
      cityName: ""
    };
  }

  /**
   * Fetches the weather info; if response.status is 200 fills the state
   * with the weather info, otherwise handles the 404 error or
   * catch a different one
   */
  getWeatherInfo = async cityName => {
    let cityWeatherInfo;
    try {
      const response = await fetchWeatherByCityName(cityName);
      if (!response || response.status !== 200) {
        throw new Error();
      }
      cityWeatherInfo = response.data;
      return { ...cityWeatherInfo, cityName };
    } catch (error) {
      console.error(error);
      return { cityName, error: true };
    }
  };

  handleChange = event => {
    this.setState({ cityName: event.target.value });
  };

  searchWeather = async cityName => {
    const cityWeatherInfo = await this.getWeatherInfo(cityName);
    this.props.handleSearch(cityWeatherInfo);
  }

  handleSubmit = event => {
    event.preventDefault();
    this.searchWeather(this.state.cityName);
  };

  render() {
    return (
      <div className={styles.searchFormContainer}>
        <form className={styles.searchForm} onSubmit={this.handleSubmit}>
          <input
            className={styles.inputSearch}
            type="text"
            placeholder="Input the city name"
            onChange={this.handleChange}
            value={this.state.cityName}
          />
          <input className={styles.searchButton} type="submit" value="Search" />
        </form>
      </div>
    );
  }
}

export default FormSearch;
