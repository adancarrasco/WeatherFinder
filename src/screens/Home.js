import React, { Component } from "react";

import MainContainer from "../components/MainContainer";
import defaultCities from "../utils/defaultCities";
import FormSearch from "../components/FormSearch";
import WeatherContainer from "../components/WeatherContainer";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      cityWeatherInfos: []
    };
  }

  componentDidMount() {
    this.initDefaultCities();
  }

  handleSearch = cityWeatherInfo => {
    this.setState({
      cityWeatherInfos: [cityWeatherInfo, ...this.state.cityWeatherInfos]
    });
  };

  initDefaultCities = () => {
    defaultCities.map(cityName => this.formSearch.searchWeather(cityName));
  };

  render() {
    return (
      <MainContainer>
        <FormSearch
          ref={obj => {
            this.formSearch = obj;
          }}
          handleSearch={this.handleSearch}
        />
        <WeatherContainer cityWeatherInfos={this.state.cityWeatherInfos} />
      </MainContainer>
    );
  }
}

export default Home;
