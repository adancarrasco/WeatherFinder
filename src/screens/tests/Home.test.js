import React from "react";
import { mount } from "enzyme";

import Home from "../Home";
import FormSearch from "../../components/FormSearch";

describe("Home screen", () => {
  it("it renders correctly", () => {
    const wrapper = mount(<Home />);
    expect(wrapper.find(FormSearch)).toHaveLength(1);
    expect(wrapper.instance().formSearch).toBeTruthy();
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
