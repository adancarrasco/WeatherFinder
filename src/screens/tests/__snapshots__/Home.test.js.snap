// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen it renders correctly 1`] = `
"<Home>
  <MainContainer>
    <div className=\\"mainContainer\\">
      <FormSearch handleSearch={[Function]}>
        <div className=\\"searchFormContainer\\">
          <form className=\\"searchForm\\" onSubmit={[Function]}>
            <input className=\\"inputSearch\\" type=\\"text\\" placeholder=\\"Input the city name\\" onChange={[Function]} value=\\"\\" />
            <input className=\\"searchButton\\" type=\\"submit\\" value=\\"Search\\" />
          </form>
        </div>
      </FormSearch>
      <WeatherContainer cityWeatherInfos={{...}}>
        <div />
      </WeatherContainer>
    </div>
  </MainContainer>
</Home>"
`;
